src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -83,6 +84,15 @@ public async Task<string> SpecializeMSISidecar(HostAssignmentContext context)
 
         public bool StartAssignment(HostAssignmentContext context)
         {
+            if (context.ZipUrl.IsScmRunFromPackage())
```

Comment:
Move this as close as possible to where the zipurl is actually used currently (ApplyContext)
For ex - 
Specialization should always respect StandbyMode and branching the code here will not do that.
There will be no log that says "Starting Assignment"
etc

Reply:
There is nothing special / unique about specialization using SCM_.. url. All existing checks are still applicable.